src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx

```diff
@@ -0,0 +1,144 @@
+---
+title: Python agent on Microsoft Azure App Services and Container Apps
+metaDescription: How to install APM for Python on Microsoft Azure App Services and Container Apps.
+redirects:
+  - /docs/agents/python-agent/hosting-services/python-agent-microsoft-azure
+  - /docs/python/python-agent-on-microsoft-azure
+freshnessValidatedDate: never
+---
+
+Learn about special considerations for using Microsoft Azure Container Apps and App Services (formerly known as Web Apps) as a hosting service with New Relic's Python agent.
+
+## What you need [#compatibility]
+
+In addition to the [Compatibility and requirements for the Python agent](/docs/agents/python-agent/getting-started/compatibility-requirements-python-agent), you'll also need the following for this tutorial:
+
+* A workng installation of Python
+* The Windows Azure SDK for Python for your platform
+
+<Callout variant="important">
+  The Python agent cannot capture telemetry for Azure Functions. Instead, you will need to use the [Azure Functions monitoring integration](/docs/infrastructure/microsoft-azure-integrations/azure-integrations-list/azure-functions-monitoring-integration/).
+</Callout>
+
+## Installing the Python agent [#install_python]
+
+To add the New Relic Python agent to your Azure application, add `newrelic` to your `requirements.txt` file.
+
+Here's an example with a sample app:
```

Comment:
Azure provides certain quickstarts in their docs for both managed services:

* [Container Apps](https://learn.microsoft.com/en-us/azure/container-apps/quickstart-portal)
* [App Service](https://learn.microsoft.com/en-us/azure/app-service/quickstart-python?tabs=flask%2Cwindows%2Cazure-cli%2Cazure-cli-deploy%2Cdeploy-instructions-azportal%2Cterminal-bash%2Cdeploy-instructions-zip-azcli)

I wonder if providing this sample code might risk it falling "out of sync" with Azure's docs. What if we were to direct customers to these quickstarts in the pre-reqs, replace this step altogether with something general (e.g., "Add the following code snippet to the [specific-file-whathaveyou] in your app", or to one of the Azure quickstarts.")

What do you think?

Reply:
My only concern with this is that those quickstarts, while convenient, utilize a lot of Behind-The-Scenes functionality within Azure.  Also, and this is my personal bias, the Azure docs are not very thorough and lack transparency.  In attempting to make the tutorials easy to follow, they often neglect to point out how to navigate common use cases for customers.  My intention here is to use a barebones example that makes the what and where of the changes needed evident.

Reply:
Awesome, I'll stick with what you have!!